src/code/z_kankyo.c

```diff
@@ -1,94 +1,2499 @@
+    GfxPrint_Destroy(&printer);
+}
+
+#define TIME_ENTRY (D_8011FB48[envCtx->unk_1F] + i)
+#define TIME_ENTRY1 (&D_8011FB48[envCtx->unk_1F][i])
+#define TIME_ENTRY2 (&D_8011FB48[envCtx->unk_20][i])
+
+void func_80075B44(GlobalContext* globalCtx);
+void func_800766C4(GlobalContext* globalCtx);
+
+#ifdef NON_MATCHING
+// Reordering in light color and fog near and far blends
+void Kankyo_Update(GlobalContext* globalCtx, EnvironmentContext* envCtx, LightContext* lightCtx, PauseContext* pauseCtx,
+                   MessageContext* msgCtx, GameOverContext* gameOverCtx, GraphicsContext* gfxCtx) {
+    f32 sp8C;
+    f32 sp88 = 0.0f;
+    u16 i;
+    u16 j;
+    s16 lightAdj;
+    u16 time;
+    EnvLightSettings* lightSettingsList = globalCtx->envCtx.lightSettingsList; // 7C
+    s32 adjustment;
+
+    if ((((void)0, gSaveContext.gameMode) != 0) && (((void)0, gSaveContext.gameMode) != 3)) {
+        func_800AA16C(globalCtx);
+    }
+
+    if (pauseCtx->state == 0) {
+        if ((globalCtx->pauseCtx.state == 0) && (globalCtx->pauseCtx.debugState == 0)) {
+            if (globalCtx->skyboxId == SKYBOX_NORMAL_SKY) {
+                globalCtx->skyboxCtx.rot.y -= 0.001f;
+            } else if (globalCtx->skyboxId == SKYBOX_CUTSCENE_MAP) {
+                globalCtx->skyboxCtx.rot.y -= 0.005f;
+            }
+        }
+
+        func_800766C4(globalCtx); // increments or decrements unk_EE[1] depending on some condition
+        func_80075B44(globalCtx); // updates bgm/sfx and other things as the day progresses
+
+        if (((void)0, gSaveContext.nextDayTime) >= 0xFF00 && ((void)0, gSaveContext.nextDayTime) != 0xFFFF) {
+            gSaveContext.nextDayTime -= 16;
+            osSyncPrintf("\nnext_zelda_time=[%x]", ((void)0, gSaveContext.nextDayTime));
+
+            if (((void)0, gSaveContext.nextDayTime) == 0xFF0E) {
+                func_80078884(NA_SE_EV_CHICKEN_CRY_M);
+                gSaveContext.nextDayTime = 0xFFFF;
+            } else if (((void)0, gSaveContext.nextDayTime) == 0xFF0D) {
+                func_800788CC(NA_SE_EV_DOG_CRY_EVENING);
+                gSaveContext.nextDayTime = 0xFFFF;
+            }
+        }
+
+        if ((pauseCtx->state == 0) && (gameOverCtx->state == GAMEOVER_INACTIVE)) {
+            if (((msgCtx->unk_E300 == 0) && (msgCtx->msgMode == 0)) || (((void)0, gSaveContext.gameMode) == 3)) {
+                if ((envCtx->unk_1A == 0) && !FrameAdvance_IsEnabled(globalCtx) &&
+                    (globalCtx->transitionMode == 0 || ((void)0, gSaveContext.gameMode) != 0)) {
+
+                    if (!((void)0, gSaveContext.nightFlag) || gTimeIncrement >= 400) {
+                        gSaveContext.dayTime += gTimeIncrement;
+                    } else {
+                        gSaveContext.dayTime += gTimeIncrement * 2; // time moves twice as fast at night
+                    }
+                }
+            }
+        }
+
+        if (((((void)0, gSaveContext.sceneSetupIndex) >= 5 || gTimeIncrement != 0) &&
```

Comment:
yeah im not sure. im not even sure what the intention is regardless of if that check was 4 or 5. like why does it being an alt header matter for syncing skybox and regular time
it might be a very specific hardcoded case. in which case i would expect a scene check or something though

Reply:
[z_bg_spot00_hanebasi](https://github.com/petrie911/oot/blob/Kankyo/src/overlays/actors/ovl_Bg_Spot00_Hanebasi/z_bg_spot00_hanebasi.c#L237) uses time stuff and sceneSetupIndex == 5, may be related idk

I'm not even sure what the purpose of skyboxTime is, is it so that the skybox isn't in a "blending" state when in areas where time is frozen like kakariko?
`func_80098D80` (Scene Command 0x10: Time Settings) would suggest that, is that the only point to it

I think it was meant to be `>= 4` and they probably mistyped as `> 4`, but since what it achieves is obscure or maybe even doesn't matter (many other things set skyboxTime) they never noticed it.

Since it's unsure and not important anyway, let's just ignore it